Test branch

evignola opened the pull request:
There was a problem with BPs data structures, due to wrong initialisation.

evignola committed:
Constants added

evignola committed:
BPs data structures successfully debugged

mstamatakis commented on Fortran90/data_types/funcv.f90:
```diff
@@ -12,18 +7,23 @@ subroutine funcv(fvec)
           v2(j)=obj_approx%eqn%rhs(i,j)
          end do
-         fvec(i)=log(obj_approx%corfun(v1,total,obj_approx))-log(obj_approx%corfun(v2,total,obj_approx))
+         fvec(i)=obj_approx%corfun(v1,total,obj_approx)-obj_approx%corfun(v2,total,obj_approx)
```
Why did you remove the logarithms here? I don't get solutions for chemical potentials higher than about -0.3 and I believe variables may be overflowing. I'd say add the logs back in.

mstamatakis commented on Fortran90/data_types/meso_approx.f90:
```diff
@@ -196,7 +244,7 @@ subroutine approx_initialisation(appr)
+     do i=7,12
       appr%hamilt%corr%intmap(i)=2
      end do
      appr%hamilt%corr%value(1)=0.1d0
```
Make a note in the code (or add an issue on Github), that this is a guess value and might need to be adjusted in the future.

mstamatakis commented on Fortran90/data_types/meso_approx.f90:
```diff
@@ -142,6 +142,52 @@ subroutine approx_initialisation(appr)
+     appr%hamilt%orig%value=0.d0
+     appr%hamilt%corr%term=0
+     appr%hamilt%corr%intmap=0
+     appr%hamilt%corr%value=0.1d0
```
Make a note in the code (or add an issue on Github), that this is a guess value and might need to be adjusted in the future.

mstamatakis commented on Fortran90/data_types/main.f90:
```diff
@@ -31,7 +31,7 @@ program main
+    
         ! Coverage vs Chemical Potential Plot
         chemp=-1.40d0
         do i=1,240
```
Let the user specify initial and final chemical potential, and the number of points (or just the Dmu). Then have the code calculate number of points.

evignola committed:
Logarithms added

evignola committed:
K2NNC1 and variable "nrows" added

evignola committed:
K2NNC2 approximation added

evignola committed:
Allocations in "funcv" removed

evignola committed:
Derivative increment in subroutine "fdjac" decreased to 1E-10. This fixes BPEC convergence

evignola committed:
h0 added as an external variable

mstamatakis committed:
Added Matlab script to plot results quickly

mstamatakis committed:
Monitor magnitude of partition function

mstamatakis committed:
Fixing algorithmic errors

Fixed wrong sign of h0: the sign of h0 in the meso_approx expressions has to opposite from that of kb*temp*log(obj_approx%part()) in main.

Also, the constant h0 has to be updated incrementally, so that it stores the free energy of the approximation at each iteration of the loop over chemical potentials.

Finally, in meso_approx the partition function and the correlation shouldn't be multiplied by exp(h0/(hb*temp)). The whole point is to avoid the overflow errors created by such large terms. This is done by redefining the reference energy h0 as the chemical potential increases.

evignola committed:
Merge pull request #6 from mstamatakis/h0

Energy constant H0

evignola committed:
Two lines of code were missing in the Newton algorithm
